roles/deploy/hooks/build-after.yml

```diff
@@ -19,7 +19,6 @@
   when:
     - composer_authentication.hostname is defined and composer_authentication.hostname != ""
     - composer_authentication.username is defined and composer_authentication.username != ""
-    - composer_authentication.password is defined and composer_authentication.password != ""
```

Comment:
In line 15:

```diff
- - arguments: --auth http-basic.{{ composer_authentication.hostname }} {{ composer_authentication.username }} {{ composer_authentication.password }}
+ - arguments: --auth http-basic.{{ composer_authentication.hostname }} {{ composer_authentication.username }} {{ composer_authentication.password | default("''") }}
```

This is because:

```sh-session
$ composer config http-basic.no-password.com my-username

In ConfigCommand.php line 782:

  Expected two arguments (username, password), got 1


config [-g|--global] [-e|--editor] [-a|--auth] [--unset] [-l|--list] [-f|--file FILE] [--absolute] [-j|--json] [-m|--merge] [--append] [--source] [--] [<setting-key> [<setting-value>...]]
```

```sh-session
// This works
$ composer config http-basic.empty-password.com my-username ''
```

Maybe ansible would complaint about `password` not found in `composer_authentication` dict as well